src/sentry/integrations/api/endpoints/data_forwarding_details.py

```diff
@@ -0,0 +1,344 @@
+                if request.access.has_project_scope(project, "project:write"):
+                    projects_to_unenroll.append(project_id)
+            except Project.DoesNotExist:
+                pass
+        return projects_to_unenroll, None
+
+    def _handle_unenroll_all_projects(
+        self, request: Request, organization: Organization, data_forwarder: DataForwarder
+    ) -> Response:
+        """
+        Unenroll all projects that the user has access to from the data forwarder.
+        Called when project_ids is an empty list.
+
+        Returns:
+            Response: 200 OK with serialized data forwarder
+        """
+        enrolled_projects = DataForwarderProject.objects.filter(
+            data_forwarder=data_forwarder
+        ).values_list("project_id", flat=True)
+
+        projects_to_unenroll, error_response = self._get_accessible_projects_to_unenroll(
+            request, organization, list(enrolled_projects)
+        )
+        if error_response:
+            return error_response
+
+        if projects_to_unenroll:
+            DataForwarderProject.objects.filter(
+                data_forwarder=data_forwarder, project_id__in=projects_to_unenroll
+            ).delete()
+
+        return Response(
+            serialize(data_forwarder, request.user),
+            status=status.HTTP_200_OK,
+        )
+
+    def _validate_project_permissions(
+        self, request: Request, projects: list[Project]
+    ) -> Response | None:
+        """
+        Validate that the user has project:write permission for all specified projects.
+
+        Returns:
+            None if user has access to all projects,
+            Response with 403 Forbidden if user lacks access to any project
+        """
+        unauthorized_projects: list[int] = []
+        for project in projects:
+            if not request.access.has_project_scope(project, "project:write"):
+                unauthorized_projects.append(project.id)
+        if unauthorized_projects:
+            return self.respond(
+                {
+                    "project_ids": [
+                        f"Insufficient access to projects: {', '.join(map(str, unauthorized_projects))}"
+                    ]
+                },
+                status=status.HTTP_403_FORBIDDEN,
+            )
+        return None
+
+    def _get_accessible_enrolled_projects(
+        self, request: Request, organization: Organization, data_forwarder: DataForwarder
+    ) -> list[int]:
+        """
+        Get the list of currently enrolled project IDs that the user has access to.
+
+        Returns:
+            List of project IDs the user has project:write access to
+        """
+        enrolled_projects = DataForwarderProject.objects.filter(
+            data_forwarder=data_forwarder
+        ).values_list("project_id", flat=True)
+
+        accessible_enrolled_projects: list[int] = []
+        projects = Project.objects.filter(organization_id=organization.id, id__in=enrolled_projects)
```

Comment:
We do these two queries quite a few times throughout the endpoint. Would it be possible to do some consolidation? (Not sure of the answer just wondering)